x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/action_stop/use_stop_action.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+
+import {
+  isDataFrameAnalyticsFailed,
+  isDataFrameAnalyticsRunning,
+  DataFrameAnalyticsListAction,
+  DataFrameAnalyticsListRow,
+} from '../analytics_list/common';
+import { stopAnalytics } from '../../services/analytics_service';
+
+import { stopActionNameText, StopActionName } from './stop_action_name';
+
+export type StopAction = ReturnType<typeof useStopAction>;
+export const useStopAction = (canStartStopDataFrameAnalytics: boolean) => {
+  const [isModalVisible, setModalVisible] = useState(false);
+
+  const [item, setItem] = useState<DataFrameAnalyticsListRow>();
+
+  const closeModal = () => setModalVisible(false);
+  const forceStopAndCloseModal = () => {
+    if (item !== undefined) {
+      setModalVisible(false);
+      stopAnalytics(item);
+    }
+  };
+
+  const openModal = (newItem: DataFrameAnalyticsListRow) => {
+    setItem(newItem);
+    setModalVisible(true);
+  };
+
+  const clickHandler = useCallback(
+    (i: DataFrameAnalyticsListRow) => {
+      if (canStartStopDataFrameAnalytics) {
+        if (isDataFrameAnalyticsFailed(i.stats.state)) {
+          openModal(i);
+        } else {
+          stopAnalytics(i);
+        }
+      }
+    },
+    [stopAnalytics]
+  );
+
+  const action: DataFrameAnalyticsListAction = useMemo(
+    () => ({
+      name: () => <StopActionName isDisabled={!canStartStopDataFrameAnalytics} />,
+      available: (i: DataFrameAnalyticsListRow) => isDataFrameAnalyticsRunning(i.stats.state),
```

Comment:
Similarly, there might need to be a check to allow force stop for jobs in a failed state.

Reply:
Good catch! I carried over the changes related to the stop action rendering but missed this one. Fixed in c9fc1b2.